### PR TITLE
Treeherder integration

### PR DESCRIPTION
**Pull request by helfi92**
This PR is to provide a way to jump to a line number, highlight line(s) and highlight error line(s) when the logviewer is embedded inside an iframe.

### CODE REVIEW

File: src/LogViewer/index.js [hunk 1]
```diff
@@ -106,6 +114,17 @@ export default class LogViewer extends React.Component {
...
       this.setState({ data: e.data });
+
+      Object.keys(e.data).forEach((name) => {
+        switch(name) {
+          case 'highlightError':
+            return this.highlightError(e.data.highlightError);
+          case 'lineNumber':
+            return this.jumpToLine(e.data.lineNumber);
+          case 'highlightStart':
+            return this.setState({ highlightStart: e.data.highlightStart, highlightEnd: e.data.highlightEnd });
```
Comment (helfi92) [inline src/LogViewer/index.js]: The state is acting weird when the logviewer is embedded inside an iframe. `this.setState` updates the state but the change is not seen unless I call setState again (send another postMessage). 
Would you happen to know why this is happening? The state seems to be updating which is why I am a little bit confused as to why it's not highlighting on the first call.

Comment (eliperelman) [reply src/LogViewer/index.js]: I could replicate this, but I made a few changes and it seems to work OK now. I'll put some notes in the review comment.

**Review by eliperelman — CHANGES_REQUESTED** [review]
I think this PR is pretty good, but I'd like to see some changes. I tested a couple things out locally to resolve some wonkiness, so let me know what you think.

First, when handling `postMessage`s in `handleMessage`, the `data` is an object containing all the state to set for the log viewer, so it's acceptable to just set all of it:

``` js
handleMessage(e) {
  if (e.data && typeof e.data === 'object') {
    this.setState(e.data);
  }
}
```

So instead of triggering functions when the state is changed here, the rest of the app should just react to a change in state. Line highlighting already works this way, so nothing to change there. I'll come back to the error stuff in a bit, so let's look at handling jumping.

Right now line jumping is done in `componentDidMount`, but now it also needs to perform those same actions when the line jumping state changes. So let's move that into it's own function:

``` js
handleJump() {
  const { lineNumber, highlightStart, jumpToHighlight } = this.state;

  if (lineNumber) {
    this.jumpToQueriedLine(lineNumber);
  } else if (jumpToHighlight && highlightStart) {
    this.jumpToQueriedLine(highlightStart);
  }
}
```

Now we can switch `componentDidMount` to use the new function:

``` js
componentDidMount() {
  this.request();
  this.handleJump();
}
```

Next, we need also trigger this jump handling when the line jumping state has changed. We can do this in a `componentDidUpdate`:

``` js
componentDidUpdate(prevProps, prevState) {
  const shouldJump = prevState.lineNumber !== this.state.lineNumber ||
    this.state.jumpToHighlight &&
    prevState.highlightStart !== this.state.highlightStart;

  if (shouldJump) {
    this.handleJump();
  }
}
```

So now whenever the component updates and either the `lineNumber` or `highlightStart` when jumping to highlight changes, we go ahead and re-trigger a jump.

---

Now, back to the error highlighting. The unified-logviewer should have no such knowledge of error highlighting, rather, just custom styling. We need to be able to change the custom styles on state changes. Also with the current implementation a new sheet is created every time, which will make overriding the changes very difficult.

There is a project called [react-helmet](https://github.com/nfl/react-helmet) which I have experimented with in a Tools branch before and worked really well. Basically we can put `<head>` changes in JSX and it will update it based on whatever. We are interested in controlling a style tag:

react-helment style tag

So what I'd like to see is that we bring in react-helmet and use it inside `render()`, which will just update based on the state. Something like:

``` jsx
import Helmet from 'react-helmet';
// ...

render() {
  // ...

  return (
    <div>
      {this.state.customStyle && <Helmet style={[{ cssText: this.state.customStyle }]} />}
      ...
    </div>
  );
}
```

---

This should allow everything to be reactive based on state. Let me know!

**Comment by helfi92** [comment]
@eliperelman You are right, we need to let the state handle everything. Implemented your changes on my end and it works like a charm. Very neat! Btw, I didn't know about react-helmet. It's awesome. Will be pushing the commit shortly.

**Comment by eliperelman** [comment]
@helfi92 sounds good! Yeah, I completely forgot about it until I realized we needed to manipulate a single `<style>` vs. constant creation, which we need React for.

**Comment by helfi92** [comment]
@eliperelman thanks again for the thorough review. Latest commit should have everything now

**Comment by eliperelman** [comment]
@helfi92 can you switch the react-helmet dep to be an exact version?

**Comment by helfi92** [comment]
@eliperelman done

**Pull request merged**